syntax = "proto3";

package smartfarm;


option go_package = "github.com/fermat01/smart-farm-iot-platform/pb;pb";


service SensorService {
  rpc StreamSensorData(stream SensorReading) returns (Ack) {}
}


message SensorReading {
  string sensor_id = 1;
  string type = 2;         
  string timestamp = 3;

  double soil_moisture = 4;
  double soil_temperature = 5;
  double air_temperature = 6;
  double humidity = 7;
  double water_tank_level = 8;
  double well_depth = 9;
  double pump_flow_rate = 10;
}


message Ack {
  string message = 1;
}
